guides/integrate-nextauth.md

```diff
@@ -0,0 +1,427 @@
+        });
+      }
+    }
+    authenticate().catch(console.error);
+  }, []);
+
+  // -- 3
+  async function biAuthenticate(url: string): Promise<string> {
+    const BeyondIdentityEmbeddedSdk = await import("@beyondidentity/bi-sdk-js");
+    let embedded = await BeyondIdentityEmbeddedSdk.Embedded.initialize();
+
+    // Display passkeys so user can select one
+    let passkeys = await embedded.getPasskeys();
+    let promptText = passkeys.map((passkey, index) => {
+      return `${index}: ${passkey.identity.username}`;
+    }).join("\n");
+    let selectedIndex = parseInt(prompt(promptText, "index")!!);
+    if (selectedIndex >= 0 && selectedIndex < passkeys.length) {
+      let selectedId = passkeys[selectedIndex].id;
+      // Perform authentication using selected id
+      let result = await embedded.authenticate(url, selectedId);
+      return Promise.resolve(result.redirectURL);
+    } else {
+      // This will fail in core as it won't match to any id
+      return Promise.resolve("unknown_id");
+    }
+  }
+
+  return (
+    <div
+      style={{
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        height: '100vh',
+      }}
+    >
+      <div className="container">
+        <div className="row">
+          <div className="d-flex justify-content-center">
+            <div className="spinner-border" role="status">
+              <span className="sr-only"></span>
+            </div>
+          </div>
+        </div>
+        <div className="row">
+          {
+            biAuthenticateResult.length > 0 &&
+            <div className="row row-cols-1 row-cols-md-1 mt-3">
+              <div className="col">
+                <code>
+                  {JSON.stringify(biAuthenticateResult, null, 2)}
+                </code>
+              </div>
+            </div>
+          }
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default BIAuthenticate;
+```
+
+</TabItem>
+<TabItem value="auth" label="Auth.js">
+
+:::warning WIP
+@auth/nextjs is work in progress. For now, please use [NextAuth.js](?nextjs=nextauth#authenticate).
+:::
+
+</TabItem>
+</Tabs>
+
+What's happening here?
+
+1. The `useEffect` is only called once on page load. In this function, we initialize the Beyond Identity SDK and use `embedded.isAuthenticateUrl` to check if the current page that was redirected to is in fact a valid `bi-authenticate` URL.
+2. If the URL is valid, we pull the URL using `window.location.href` and pass that directly into `biAuthenticate` in step 3.
+3. `biAuthenticate` calls `embedded.authenticate` with a valid `bi-authenticate` URL. This function performs a challenge/response against a passkey bound to your browser. Note that the callback in `embedded.authenticate` contains logic in order to prompt a user to select a passkey if there is more than one.
+4. Finally, the response of `embedded.authenticate` contains a `redirectURL`. Follow this redirectURL to complete the OAuth/OIDC flow.
```

Comment:
Same thing, here to keep this under the NextAuth tab

Reply:
Makes sense to me. Done.